airflow/hooks/hive_hooks.py

```diff
@@ -740,6 +739,7 @@ class HiveServer2Hook(BaseHook):
     """
     def __init__(self, hiveserver2_conn_id='hiveserver2_default'):
         self.hiveserver2_conn_id = hiveserver2_conn_id
+        self._cursor_description = []
```

Comment:
IMO we don't need to expose the cursor concept at the class level--it can also be not thread safe if we us and reset ourselves (this part I need to read a bit more on python myself to be sure).

Reply:
If the cursor can be changed to a local variable, that would be better of course.

Personally I don't see how multiple threads use the same hook, or this might be implemented by hand.

Reply:
I don't think there will be such case neither. The only thing I can think of is that people making such calls in their python callback. This raises the question that do we want people to use the hook directl (people in Airbnb is doing this a lot)?